src/Backoffice/Task/Domain/Actions/UpdateTaskAction.php

```diff
@@ -0,0 +1,37 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Lightit\Backoffice\Task\Domain\Actions;
+
+use Lightit\Backoffice\Task\App\Notifications\TaskAssignmentNotification;
+use Lightit\Backoffice\Task\Domain\DataTransferObjects\TaskDto;
+use Lightit\Backoffice\Task\Domain\Models\Task;
+
+class UpdateTaskAction
+{
+    public function execute(Task $task, TaskDto $dto): Task
+    {
+        $previousEmployeeId = $task->employee_id;
+        $previousEmployee = $task->employee;
+
+        $task->fill($dto->toArray());
+
+        if ($task->isDirty('employee_id') && $previousEmployeeId !== $task->employee_id) {
+            // notify the newly assigned employee
+            $newEmployee = $task->employee()->firstOrFail();
+
+            $newEmployee->notify(
+                new TaskAssignmentNotification(
+                    task: $task,
+                    isReassignment: true,
+                    previousEmployee: $previousEmployee
+                )
+            );
+        }
+
+        $task->saveOrFail();
+
+        return $task;
+    }
```

Comment:
```suggestion
public function execute(Task $task, TaskDto $taskDto): Task
    {
        $previousEmployeeId = $task->employee_id;

        $task->updateOrFail([
            'title' => $taskDto->title,
            'description' => $taskDto->description,
            'status' => $taskDto->status,
            'employee_id' => $taskDto->employee->id,
        ]);

        $this->notifyNewEmployee($task, $previousEmployeeId);

        return $task;
    }
    
    private function notifyNewEmployee(Task $updatedTask, int $previousEmployeeId): void
    {
        if ($previousEmployeeId !== $updatedTask->employee_id) {
            $updatedTask->employee->notify(new TaskAssignmentNotification($updatedTask));
        }
    }
```
A little bit more clean. Also, do not leave comments within the code. It should be self explanatory.
Also, make sure you're using captain hook as I think this should be captured by it